overlore/sqlite/db.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import annotations
+
+import os
+from sqlite3 import Connection
+from threading import Lock
+from typing import Any, Callable
+
+import sqlean
+
+PreloadFunction = Callable[[Connection], None]
+FunctionCallable = Callable[..., Any | None]
+CustomFunction = tuple[str, int, FunctionCallable]
+
+
+class Database:
+    db: Connection
+    lock = Lock()
+
+    def _lock(self) -> None:
+        self.lock.acquire(blocking=True, timeout=1000)
+
+    def _release(self) -> None:
+        self.lock.release()
+
+    def __init__(self) -> None:
+        raise RuntimeError(f"{self.__class__.__name__}: call instance() instead")
+
+    def _load_extensions(self, extensions: list[str]) -> None:
+        for ext in extensions:
+            self.db.load_extension(ext)
+
+    def _insert(self, query: str, values: tuple[Any, ...]) -> int:
+        self._lock()
```

Comment:
can we sanity check whether we need locks on inserts?

https://ricardoanderegg.com/posts/python-sqlite-thread-safety/ 

> Serialized. In serialized mode, SQLite can be safely used by multiple threads with no restriction. The default mode is serialized.

we may need to ensure we initialize the sqlitedb from the bindings, assuming that we have a default-mode sqlite instance

```
conn = sqlite3.connect(<db path>, check_same_thread=False)
```

Reply:
Looks like it initialises the DB threadsafe by default. I'll add tests for this in this PR to ensure we are a-ok